### PR TITLE
Simplifying/refactoring the FieldFactory

### PR DESCRIPTION
**Pull request by luizfelippesr**
The current FieldFactory base class would probably work better as a regular class which takes the target `field_class`, `active_parameters`, `priors` and `default_parameters` as arguments.  

Some other adjustments also need to be made: currently, one has always to specify priors and default values for all parameters, instead of priors for active parameters and default values for inactive ones. 

This PR addresses these issues.

### CODE REVIEW

**Comment by 1313e** [comment]
You will still have to refactor the code to implement standard field factories for all fields.

**Comment by luizfelippesr** [comment]
> You will still have to refactor the code to implement standard field factories for all fields.

And the docs... I will get there...

**Comment by 1313e** [comment]
Why have you not created any standard classes?

**Comment by luizfelippesr** [comment]
> Why have you not created any standard classes?

I thought it was easier and more practical just keeping the previous syntax for specialised FieldFactories (e.g. the ones in fields.hamx) and adjusting things slightly so that a FieldFactory for a new field could be done wtihout the requirement of sub-classing (see: https://github.com/IMAGINE-Consortium/imagine/pull/77/commits/cc21b1ef9ed25e6e94b0b12f9eb0322e84c8a601 and https://github.com/IMAGINE-Consortium/imagine/pull/77/commits/6346b85db4a004c6a758571281fe8202aaa53652 ).

After some thought, I realised the previous "shell-style" FieldFactory subclass were a very practical way of setting up default settings for frequently used Fields. However, it was very annoying being forced to make a new subclass when experimenting with a new Field. hence, the present compromise.

Unless I misunderstand what you mean by "standard classes"...

**Comment by 1313e** [comment]
But you didn't use the previous syntax.
Previously, we already had pre-made standard factory classes.
Now we don't.

File: imagine/tests/test_templates.py [hunk 1]
```diff
@@ -197,11 +175,20 @@ def test_pipeline_template():
...
+        field_class=FakeRandomTE,
+        grid=grid,
+        active_parameters=['param'],
+        priors={'param': img_priors.FlatPrior(xmin=0, xmax=10.)})
+
+    B_factory = img_fields.FieldFactory(
+      field_class=img_fields.ConstantMagneticField,
+      grid=grid,
+      active_parameters=['Bx', 'By'],
+      default_parameters={'Bz': 3.*muG},
```
Comment (shutsch) [inline imagine/tests/test_templates.py:188]: Just out of interest, why did you fix the Bz value here, while it was inferred previously?

**Review by shutsch — APPROVED** [review]
Nice work and also a lot of improvements in the documentation. No major objections from my side.

**Review by 1313e — APPROVED** [review]

File: imagine/tests/test_templates.py (hunk 1 above)
Comment (luizfelippesr) [reply imagine/tests/test_templates.py:188]: The idea was testing both the infrastructure for active and inactive parameters. Arbitrarily, I made Bx and By active, and Bz inactive.

File: doc/source/components.rst [hunk 2]
```diff
@@ -327,26 +327,49 @@ the Simulator, which computes simulated Observables for comparison
...
 
-.. literalinclude:: ../../imagine/templates/field_factory_template.py
 
-The object `Prior A` must be an instance of
+The object `YourPriorChoice` must be an instance of
 :py:class:`imagine.priors.Prior <imagine.priors.prior.Prior>`
-(see section `Priors`_ for
-details). A flat prior (i.e. a uniform, where all parameter values are
+(see section `Priors`_ for details).
+A flat prior (i.e. a uniform, where all parameter values are
```
[the remark this answers is not in the document]
Comment (luizfelippesr) [reply doc/source/components.rst:345]: Thanks, I'll fix it in the next commit.

**Pull request merged**